### PR TITLE
Prefix test title with test location

### LINKED ISSUE #53: Prefix test results with filename when running multiple files
We should prefix the tests with the filename minus extension and prefix `test-` so [this](https://github.com/sindresorhus/got/pull/104/files#diff-4b237e7473a059a1f82a01d75d69c5f1R26) isn't needed.

`test-url.js` with `test('foo bar')` would be:

```
✔ url - foo bar
```

-

// @floatdrop

### PR DESCRIPTION
**Pull request by vdemedes**
Fixes #53 

Uses `figures.pointerSmall` to replace `/` and as a separator between prefix and test title.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -43,12 +45,24 @@ function error(err) {
 function test(data) {
 	var isError = data.err.message;
 
+	var prefix = '';
+
+	if (fileCount > 1) {
+		var separator = ' ' + figures.pointerSmall + ' ';
+
+		prefix = data.file
+			.replace(process.cwd() + '/', '')
```
Comment (Qix-) [inline cli.js]: Should use `path.relative()` here.

File: cli.js [hunk 2]
```diff
@@ -43,12 +45,24 @@ function error(err) {
...
+			.replace(/test\-|test\//g, '')
```
Comment (Qix-) [inline cli.js]: If the directory has `test-` then that's a problem. If they're on windows, `\/` might be a problem.

File: cli.js [hunk 3]
```diff
@@ -43,12 +45,24 @@ function error(err) {
...
+			.replace(/\.js$/, '')
```
Comment (Qix-) [inline cli.js]: Use `path.basename(..., '.js')`.

File: cli.js [hunk 4]
```diff
@@ -43,12 +45,24 @@ function error(err) {
...
+			.replace(/\//g, separator) + separator;
```
Comment (Qix-) [inline cli.js]: Use `path.sep` here.

**Comment by vdemedes** [comment]
@Qix- all good points, I should not submit PRs after 10pm. Will review this again tomorrow, thanks!

File: cli.js [hunk 5]
```diff
@@ -33,6 +34,7 @@ var cli = meow({
 	string: ['_']
 });
 
+var fileCount = 0;
```
Comment (Qix-) [inline cli.js]: Is this going to pose some sort of problem with concurrency? // @sindresorhus

Comment (vdemedes) [inline cli.js]: No issues here, since it is in `cli.js`. This is an entrypoint file.

Comment (Qix-) [inline cli.js]: :+1: disregard then.

**Comment by Qix-** [comment]
Haha no problem; common path stuff.

**Comment by vdemedes** [comment]
Added changes to `babel.js` to detect if generators are supported. If they are supported, babel's gonna use `asyncToGenerator` transform and blacklist `regenerator` transform. Before that, tests on 0.10 and 0.12 failed, because they don't support generators.

**Comment by Qix-** [comment]
Lgtm :+1:  @sindresorhus will be the ultimate decider.

File: cli.js [hunk 6]
```diff
@@ -40,15 +42,31 @@ function error(err) {
 	process.exit(1);
 }
 
+function prefixTitle(file) {
+	var separator = ' ' + figures.pointerSmall + ' ';
```
Comment (sindresorhus) [inline cli.js]: `var separator = ' ' + chalk.dim(figures.pointerSmall) + ' ';`

**Comment by sindresorhus** [comment]
The prefixing doesn't seem to work when specifying a direct path:

```
$ ava test/foo/http.js

  ✔ test title
```

Expected:

```
$ ava test/foo/http.js

  ✔ foo › http › test title
```

**Comment by vdemedes** [comment]
@sindresorhus This PR is related to issue #53, which is "Prefix test results with filename when running multiple files". So this output is expected. When running a single test file, I don't think we need prefixes, since direct path is specified.

**Comment by Qix-** [comment]
@vdemedes while I agree that that is sound behavior, I'm with @sindresorhus on it; consistency is important.

**Comment by sindresorhus** [comment]
Oh right. That makes sense. If only one file, it should not show the path.

**Comment by sindresorhus** [comment]
Super work @vdemedes. I really like this feature! :)

**Comment by sindresorhus** [comment]
tumblr_lxx3edkjhr1qdlh1io1_400

**Comment by Qix-** [comment]
That gif is life... :baby:

**Pull request merged**